charts/seed-controlplane/charts/kube-controller-manager/templates/kube-controller-manager-networkpolicy.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: {{ include "networkpolicyversion" . }}
+kind: NetworkPolicy
+metadata:
+  name: kube-controller-manager-allow-metadata-service
+  namespace:  {{ .Release.Namespace }}
+spec:
+  podSelector:
+    app: kubernetes
+    role: controller-manager
+  policyTypes:
+  - Egress
+  egress:
+  - to:
+    - ipBlock:
+        cidr: 169.254.169.254/32
```

Comment:
You will also have to whitelist atleast "apiserver" and DNS. I am not sure if at all any other egress are required. If required, they should also be listed here.

Reply:
This is whitelist of traffic. There is no blacklist policy that prevents `kube-controller-manager` from talking to the `kube-apiserver`

Reply:
That's how `NetworkPolicy` egress rule works. If at all there is single egress rule whitelisting specific cidr. It will block all IP ranges out of that CIDR.  You can verify this yourself. Apply the above `NetworkPolicy` to any of `kube-controller-manager` in shoot's control-plane and restart the `kube-controller-manager`. It will loose the leader-election since it is not able to talk to `apiserver`. Even try `exec` into container, `ping` will result in packet loss.

Reply:
You can't `ping` a K8S `Service` as it only supports `tcp` and `udp`. `imcp` is not supported - you can't do a `ping kubernetes.default.svc.cluster.local` on any cluster

Reply:
Agreed. But that's not what i meant. 
Try this:
* ping google.com : Probably irrelevant. Won't work 
* ping < apiserver-pod-cluster-ip >: won't work.
* Try creating any deployemt/pod on shoot. It will not reconcile.